### ISSUE TITLE
Uninstalling/Disabling Patches

### ISSUE THREAD

**Issue by HalfEnder776** [marcos4503/ts3-dream-launcher#11]
I can understand that uninstalling some patches like the GPU/CPU ones would be permanent but I don't understand why we cannot disable patches like naars story progression or other mods? Especially since mods like naars story progression can cause lag....

**Comment by HalfEnder776** [comment]
Also is there a log file anywhere? The launcher crashes sometimes and I am not sure why.

**Comment by HalfEnder776** [comment]
Btw of of the installed patch mods is called #ERROR-UNKOWN-PATCH-MOD#, I assume this is a bug?

**Comment by marcos4503** [comment]
Regarding the advanced mode option, there are already some things more geared towards advanced users. For example, in the Mods list, an advanced user can open a Mod in S3PE to edit it more deeply. This can be useful if the user wants to translate it, or even adjust or remove features. I think the most restrictive part is Patches. I'll see what I can do about that.

Regarding Crashes, there are some logs focused on only a few features, since most features have errors that are easier to spot. In which area is the Launcher crashing? Could it be during startup?

Regarding the "#ERROR-UNKOWN-PATCH-MOD#" error, this error probably occurs because one of the mods installed through Patches was removed, or was renamed to a name other than the default, which makes it impossible for the Launcher to detect which mod it is, since it can't find it in the internal mod name DB.

**Comment by HalfEnder776** [comment]
Thanks for taking my suggestion into consideration.

As for the crashes I was getting crashes when double clicking on the hex values provided during the conflict stage of the custom mod installer. I was getting a startup crash later on (not sure why) and since there were no logs I had to do a reinstall.

I don't think I touched anything in the patches part of the packages folder so I have no clue where "#ERROR-UNKOWN-PATCH-MOD#" came from.

Also I would like to report that some of the launcher settings do not seem to work. When I launch the sims 3 High Detail Lots gets set to 4 and Enable Object Hiding gets set to false no matter what I set ingame or in the launcher. The game also asks me if I want to enable Edge scrolling and if I want to log in every time I start it.

**Comment by marcos4503** [comment]
I see. It's likely that the Launcher is having trouble accessing the files in your computer's "Documents/EA/The Sims 3" folder. All of the Launcher settings should be applied without issue.

Check to see if your computer has OneDrive installed. If so, check to see if the "Documents" folder path has been changed from "C:/Users/user/Documents" to something like "C:/Users/user/OneDrive/Documents".

If the Documents folder path is okay, one thing you can try is to Save your preferences using the Launcher, then reset the Launcher's settings template. There should be a button for this in the "Launcher" tab in Settings.

As for the Patches error, close the Launcher, then try accessing your game's "Mods" folder, then go to "Packages/DL3-Patches" and delete all of the folder's contents. Open the Launcher and reinstall all of the Patches. Maybe this will fix the problem. In fact, the crash when clicking on the HEX in the mod installation window may be caused by this.

**Comment by HalfEnder776** [comment]
Yep my documents folder is just "C:\Users\User\Documents". Also I tested it and it seems the launcher is what causes my settings to reset. They save perfectly fine unless I reopen the launcher or click the save button. My personal guess is that it might have something to do with having both a dedicated and integrated graphics card, I tried setting all the launcher executables to use my dedicated gpu but that didn't help :/

As for the crash with the hex values, no that should be impossible the patches error is appearing on a fresh install where I removed the old "C:\Users\User\Documents\Electronic Arts\The Sims 3" folder. Also I figured out what is causing the patches error. It was "FasterElevatorMoving.package.disabled". For some reason it is disabled I have no clue why I swear I never touched it....

**Comment by marcos4503** [comment]
It is important to keep in mind that in-game settings are not retained by the Launcher. Whenever the Launcher is opened, it will overwrite the settings using the settings defined in it. This design is intended to provide a more reliable way of maintaining your settings, as the game may reset its settings on occasion (such as when updating drivers). Now, as for the different GPUs, that could be the reason, but I think it's unlikely to be the case. Everyone I know who uses the Launcher and has both an integrated and a dedicated GPU doesn't have this problem. (including me)

Regarding the Patches error... That may indeed be the problem... Some of the mods installed by Patches are disabled by the Launcher if you don't have an Expansion Pack required for the mod. In that case, the different format of the mod may be causing the problem... So, it does indeed seem to be a bug. I'll check this out as well when I work on the next Launcher update.

Regarding the Crash error when trying to access the HEX from the installation window...
What mods do you have installed in your game? 
What Launcher Patches are installed?
And finally, what is the mod you are trying to install that is causing the conflict warning that causes the crash?

**Comment by HalfEnder776** [comment]
Yes I understand why the launcher handles the settings I just wanted to clarify that the launcher seems to be the problem not the game. (In case that helps you)

As for the hex crash, from what I have seen it seems to happen for any conflicting mod.
However if it helps I first tried to install clean ui after already installing gamefreaks loading screen overhaul. This was via a manual installation since I did not have any expansions installed at the time and despite the requirements those mods work fine without them. As for patches it should be all of them except for the tuning one.

**Comment by marcos4503** [comment]
I see. I'll add this HEX crash to my list of things to check in the next update.

About the preferences error... Could you post here the "options.ini" file present in your The Sims 3 folder, after starting the Launcher?

Then, this same "options.ini" file, but after starting the game and loading a neighborhood, closing the game and closing the Launcher? (don't open the Launcher again, before share the file)

The Template file, present in "<launcher_installation_directory>/Content/options-template.ini", would also be essential.

**Comment by marcos4503** [comment]
With these files I can try to find inconsistencies. It doesn't look like a Launcher script error or anything like that, so it could be something the game is rejecting...

**Comment by HalfEnder776** [comment]
[Debug-Sims3.zip](https://github.com/user-attachments/files/17894241/Debug-Sims3.zip)

**Comment by marcos4503** [comment]
I forgot to ask, but could you also run "dxdiag" on your computer, click "Save Information" and post the resulting log here?

**Comment by HalfEnder776** [comment]
[DxDiag.txt](https://github.com/user-attachments/files/17911430/DxDiag.txt)

**Comment by marcos4503** [comment]
Thank you very much for all this information. I will review it while creating the next version and try to find the best and most reliable way to fix these issues. :)

**Comment by marcos4503** [comment]
An official guide has been created with instructions on how to uninstall Patches. There is also a warning about this within the Launcher, in version 1.5.0.
